packages/file-search/src/browser/quick-file-open.ts

```diff
@@ -29,18 +38,49 @@ export class QuickFileOpenService implements QuickOpenModel {
     }
 
     private wsRoot: FileStat | undefined;
+    private showIgnored: boolean = false;
+    private opened: boolean = false;
 
     isEnabled(): boolean {
         return this.wsRoot !== undefined;
     }
+    isOpened(): boolean {
+        return this.opened;
+    }
+    isShowIgnored(): boolean {
+        return this.showIgnored;
+    }
 
     open(): void {
-        this.quickOpenService.open(this, {
-            placeholder: 'file name to search',
-            fuzzyMatchLabel: true,
-            fuzzyMatchDescription: true,
-            fuzzySort: true
-        });
+        let placeholderText = "";
+        placeholderText = 'file name to search Press ' + this.getkeyCommand() + ' to toggle Show / Hide .gitignore files';
+        this.showIgnored = !this.isShowIgnored();
```

Comment:
You should only toggle `showIgnored` in the `else` scope below.  We only want to toggle it when the user presses ctrl-p while the dialog is open.  This is the counterpart to the fact that the condition is reversed in `file-search-service-impl.ts`.

Reply:
Done